test/tools/TestExe/TestExe.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Threading;
+using System.Diagnostics;
+
+namespace TestExe
+{
+    class TestExe
+    {
+        static void Main(string[] args)
+        {
+            if (args.Length > 0)
+            {
+                switch(args[0])
+                {
+                    case "-echoargs":
+                        for (int i = 1; i < args.Length; i++)

```

Comment:
Maybe put the test code in each case in a function if we plan to add multiple tests?
And it would be good to add comments describing the purpose of the test code.